src/coreclr/jit/codegencommon.cpp

```diff
@@ -5285,8 +5285,10 @@ void CodeGen::genFnProlog()
 #endif // TARGET_ARMARCH
 
 #if defined(TARGET_XARCH)
+    genClearAvxStateInProlog();
+
     // Preserve callee saved float regs to stack.
-    genPreserveCalleeSavedFltRegs(compiler->compLclFrameSize);
+    genPreserveCalleeSavedFltRegs();
```

Comment:
Any reason why you call it before rather than just as part of `genPreserveCalleeSavedFltRegs`?

I don't expect us to get more calls to preserve registers in the future, but seems like something that "could" be grouped

Reply:
I have done it this was so that the flow of genFnProlog/genFnEpilog is more similar to the flow of genFuncletProlog/Epilog.

The first iteration of the fix had just called genPreserveCalleeSavedFltRegs/genRestoreCalleeSavedFltRegs with RBM_NONE from genFuncletProlog/Epilog, but that looked odd since the funclet prolog/epilog won't actually save/restore any floating-point registers. It made me to factor it out to a separate method.